### PR TITLE
Show screenshot filename multiple lines

### LINKED ISSUE #466: [IDE Plugin] Show multiple lines when screenshot filename is long
When we perform BDD or variation tests for configurations, screenshot filenames can be lengthy. However, it's crucial to be able to see the entire filename. Therefore, it would be beneficial to have the ability to view the complete filename, even if it spans multiple lines.
You can debug the IDE plugin by running the `runIde` Gradle task.

### PR DESCRIPTION
**Pull request by sanao1006**
Close #466 

- Get the maximum number of characters of text that will fit in the screen width with a binary search, split the string, and concatenate them with `<br>` tag.
- May be slightly inaccurate.

### video

https://github.com/user-attachments/assets/17354551-8382-4e9f-b69d-8a3111fd3978

### CODE REVIEW

File: roborazzi-idea-plugin/src/main/kotlin/com/github/takahirom/roborazzi/idea/preview/RoborazziPreviewTool.kt [hunk 1]
```diff
@@ -241,6 +241,56 @@ class ImageListCellRenderer : ListCellRenderer<Pair<String, Long>> {
     val isSelected: Boolean
   )
 
+  private fun collapseFileNameToFitWidth(text: String, maxWidth: Int): String {
+    val label = JBLabel(text)
+    val fontMetrics = label.getFontMetrics(label.font)
+
+    // Adjust the maximum width to take into account the width of the scrollbar (about 20 pixels).
+    val bufferedWidth = maxWidth - 20
```
Comment (takahirom) [inline roborazzi-idea-plugin/src/main/kotlin/com/github/takahirom/roborazzi/idea/preview/RoborazziPreviewTool.kt:249]: Thanks. I'm not sure if this is related to this pull request, but my IDE froze when I was shrinking the tool window.
image

Comment (sanao1006) [reply roborazzi-idea-plugin/src/main/kotlin/com/github/takahirom/roborazzi/idea/preview/RoborazziPreviewTool.kt:249]: Hmm...
Excuse me, is this a screenshot test of the Doroidkaigi2024 project?
In my environment, it's  `./gradlew runIde` to start intellij and select `conference-app-2024`, I get an AGP compatibility error.
Are you debugging the Roborazzi plugin in some other way?

Comment (sanao1006) [reply roborazzi-idea-plugin/src/main/kotlin/com/github/takahirom/roborazzi/idea/preview/RoborazziPreviewTool.kt:249]: Thanks!
I could reproduce it.
You are right, the “-20” could be doing something wrong...🤔
I'll try to make some more improvements.

File: roborazzi-idea-plugin/src/main/kotlin/com/github/takahirom/roborazzi/idea/preview/RoborazziPreviewTool.kt [hunk 2]
```diff
@@ -241,6 +241,59 @@ class ImageListCellRenderer : ListCellRenderer<Pair<String, Long>> {
...
+    /**
+     * Performs binary search to find the maximum number of characters that fit within the specified width.
+     * This helps to determine the end position of the substring that fits within the available width.
+     */
+    fun findMaxCharsForWidth(start: Int): Int {
+      var low = start
+      var high = minOf(text.length, start + bufferedWidth)
+
+      while (low < high) {
+        val mid = (low + high + 1) / 2
```
Comment (takahirom) [inline roborazzi-idea-plugin/src/main/kotlin/com/github/takahirom/roborazzi/idea/preview/RoborazziPreviewTool.kt:268]: 👍

**Comment by sanao1006** [comment]
I did it!!!
Thanks for your review!!

**Pull request merged**